src/Chapter3.hs

```diff
@@ -971,6 +1084,28 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday 
+  = Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday
+  deriving (Show, Eq, Ord, Enum)
+
+isWeekend :: Weekday -> Bool
+isWeekend Saturday = True
+isWeekend Sunday = True
+isWeekend _ = False
+
+nextDay :: Weekday -> Weekday
+nextDay Sunday = Monday
+nextDay d = succ d
+
+daysToParty :: Weekday -> Integer
+daysToParty Friday = 0
+daysToParty d = 1 + daysToParty (nextDay d)
```

Comment:
Very creative solution! 🧠 📆 

This also could be implemented using the `Enum` instance of `Weekday`, `mod` function and knowledge, that we have 7 days in a week 🙂